recipes/perl/assets/create_assetGroup.pl

```diff
@@ -0,0 +1,205 @@
+
+    my $resp = $ua->request($req);
+    if ($resp->is_success) {
+        my $message = decode_json($resp->content);
+        $token = $message->{"token"};
+        print "Login succeeded with status code: ", $resp->code, "\n";
+    }
+    else {
+        print "Request failed with status code: ", $resp->code, "\n";
+    }
+    return $token;
+}
+
+# subroutine to get assets
+sub create_assetGroup {
+    my $url = "$base_url/asset-service/queries";
+
+    my $asset_data = qq({
+  "data": {
+    "type": "query",
+    "attributes": {
+      "queryName": "create-or-update-assets",
+      "workloads": ["vmware"],
+      "parameters": {
+        "objectList": [
+        {
+            "correlationId": "corr-groupvcfdf",
+            "type": "vmwareGroupAsset",
+            "assetGroup": {
+              "description": "sampleDescription",
+              "assetType": "vmGroup",
+              "filterConstraint": "sampleFilterConstaint",
+              "oDataQueryFilter": "commonAssetAttributes/displayName eq 'test445'",
+              "commonAssetAttributes": {
+                "displayName": "sampleGroup249",
+                "workloadType": "vmware",
+                "protectionCapabilities": {
+                  "isProtectable": "YES",
+                  "isProtectableReason": "sampleReason",
+                  "isRecoverable": "NO",
+                  "isRecoverableReason": "sampleReason"
+                },
+                "detection": {
+                  "detectionMethod": "MANUAL"
+                }
+              }
+            }
+          }
+        ]
+      }
+    }
+  }
+});
+
+    my $req = HTTP::Request->new(POST => $url);
+    $req->header('Content-Type' => $content_type);
+    $req->header('Authorization' => $token);
+    $req->content($asset_data);
+
+    print "\n\n**************************************************************";
+    print "\n\n Making POST Request to create assetGroup\n\n";
+
+    my $resp = $ua->request($req);
+ 
+        my $message = $resp->decoded_content;
+        print "Create vmware Asset Group succeeded with status code: ", $resp->code, "\n\n\n";
+        print "=======.================\n";
+
+        my $json = decode_json($message);
+        my $id = $json->{'data'}->{'id'};
+        
+        my $url = "$base_url/asset-service/queries/$id";
+        my $req = HTTP::Request->new(GET => $url);
+        $req->header('Accept' => $content_type);
+        $req->header('Authorization' => $token);
+
+        print "\n\n**************************************************************";
+        print "\n\n Making GET Request for the QueryID response\n\n";
+
+        my $resp = $ua->request($req);
+        my $message = $resp->decoded_content;
+
+        my $json = decode_json($message);
+        my @response = @{$json->{'data'}};
+        my $response = $_;
+        my @workItem = @{$response->{'attributes'}->{'workItemResponses'}};
+        my $workItem = $_;
```

Comment:
What does `$_` refer to here? We're not in a loop.

Would you please not use variables with the same name but different sigils?

Reply:
even the workitem's is also an array.
here is the sample example output response:

"data": [
    {
      "type": "OperationResponse",
      "id": "VMware_create-or-update-assets",
      "attributes": {
        "status": "SUCCESS",
        "percentComplete": 100,
        "totalWorkItems": 1,
        "totalSuccessfulWorkItems": 1,
        "totalFailedWorkItems": 0,
        "totalInProgressWorkItems": 0,
        "workItemResponses": [
          {
            "correlationId": "corr-",
            "statusDetails": {
              "status": "SUCCESS",
              "message": "CREATED",
              "percentComplete": 100
            },
            "links": {
              "self": {
                "href": "/asset-service/workloads/vmware/assets/e3da1a0e-b37f-4519-ab61-5d31adeb5a9f"
              }
            }
          }
        ]
      }

let me know if am missing anything here.